x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -1119,14 +1119,14 @@ public AsyncSender interceptSender(AsyncSender sender) {
 
     @Override
     public UnaryOperator<RestHandler> getRestHandlerWrapper(ThreadContext threadContext) {
-        if (enabled == false) {
-            return null;
+        boolean extractClientCertificate = false;
+        if (enabled && HTTP_SSL_ENABLED.get(settings)) {
+            final SSLConfiguration httpSSLConfig = getSslService().getHttpTransportSSLConfiguration();
+            extractClientCertificate = getSslService().isSSLClientAuthEnabled(httpSSLConfig);
         }
```

Comment:
Nit: we can save the extra `finalExtractClientCertificate` variable.

```suggestion
        final boolean extractClientCertificate;
        if (enabled && HTTP_SSL_ENABLED.get(settings)) {
            final SSLConfiguration httpSSLConfig = getSslService().getHttpTransportSSLConfiguration();
            extractClientCertificate = getSslService().isSSLClientAuthEnabled(httpSSLConfig);
        } else {
            extractClientCertificate = false;
        }
```